spring-vault-core/src/main/java/org/springframework/vault/core/lease/domain/RequestedSecret.java

```diff
@@ -95,12 +109,24 @@ public String toString() {
 		/**
 		 * Renew lease of the requested secret until secret expires its max lease time.
 		 */
-		RENEW,
+		RENEW {
+			@Override
+			public RequestedSecret buildRequestedSecret(String path) {
+				return renewable(path);
+			}
+		},
 
 		/**
 		 * Renew lease of the requested secret. Obtains new secret along a new lease once
 		 * the previous lease expires its max lease time.
 		 */
-		ROTATE;
+		ROTATE {
+			@Override
+			public RequestedSecret buildRequestedSecret(String path) {
+				return rotating(path);
+			}
+		};
+
+		abstract RequestedSecret buildRequestedSecret(String path);
```

Comment:
Introduction of the new API methods creates a cycle to `RequestedSecret`. How about keeping the whole construction inside `buildFromMode(…)`?